cmd/juju/commands/debughooks.go

```diff
@@ -68,56 +75,102 @@ func (c *debugHooksCommand) Init(args []string) error {
 }
 
 type charmRelationsAPI interface {
-	CharmRelations(serviceName string) ([]string, error)
+	CharmRelations(applicationName string) ([]string, error)
 }
 
-func (c *debugHooksCommand) getServiceAPI() (charmRelationsAPI, error) {
+type ActionsAPI interface {
+	ApplicationCharmActions(params.Entity) (map[string]params.ActionSpec, error)
+}
+
+func (c *debugHooksCommand) getApplicationAPI() (charmRelationsAPI, error) {
 	root, err := c.NewAPIRoot()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	return application.NewClient(root), nil
 }
 
-func (c *debugHooksCommand) validateHooks() error {
+func (c *debugHooksCommand) newActionsAPI() (ActionsAPI, error) {
+	root, err := c.NewAPIRoot()
+	if err != nil {
+		return nil, err
+	}
+	return action.NewClient(root), nil
+}
+
+func (c *debugHooksCommand) validateHooksOrActions() error {
 	if len(c.hooks) == 0 {
 		return nil
 	}
-	service, err := names.UnitApplication(c.Target)
+	appName, err := names.UnitApplication(c.Target)
 	if err != nil {
 		return err
 	}
-	serviceAPI, err := c.getServiceAPI()
+
+	// Get a set of valid hooks.
+	validHooks, err := c.getValidHooks(appName)
 	if err != nil {
 		return err
 	}
-	relations, err := serviceAPI.CharmRelations(service)
+
+	// Get a set of valid actions.
+	validActions, err := c.getValidActions(appName)
 	if err != nil {
 		return err
 	}
 
-	validHooks := make(map[string]bool)
+	// Is passed argument a valid hook or action name?
+	// If not valid, err out.
+	allValid := validHooks.Union(validActions)
+	for _, hook := range c.hooks {
+		if !allValid.Contains(hook) {
+			logger.Infof("unknown hook or action %s, valid hook or action names: %v", hook, allValid.SortedValues())
+			return errors.Errorf("unit %q does not contain hook nor action %q", c.Target, hook)
```

Comment:
Sounds weird. I think this is easier to read: `unit %q contains neither hook nor action %q`